src/ctapipe/io/interpolation.py

```diff
@@ -0,0 +1,283 @@
+from abc import ABCMeta, abstractmethod
+from typing import Any
+
+import astropy.units as u
+import numpy as np
+import tables
+from astropy.time import Time
+from scipy.interpolate import interp1d
+
+from ctapipe.core import Component, traits
+
+from .astropy_helpers import read_table
+
+
+class StepFunction:
+
+    """
+    Step function Interpolator for the gain and pedestals
```

Comment:
Some more description would be useful here.  What is meant by "step function"? I guess you mean it is similar to the "nearest" option in `interp1d`, but only backward in time? (e.g. only the nearest previous measurement?)